tests/common/plugins/sanity_check/checks.py

```diff
@@ -241,6 +248,159 @@ def check_monit_services_status(check_result, monit_services_status):
 
     return check_result
 
+def get_arp_pkt_info(dut, intf):
+    intf_mac = dut.interface_facts()['ansible_facts']['ansible_interface_facts'][intf]['macaddress']
+    mgmt_ipv4 = None
+
+    mgmt_intf_facts = dut.get_running_config_facts()['MGMT_INTERFACE']
+
+    for mgmt_intf in mgmt_intf_facts:
+        for mgmt_ip in mgmt_intf_facts[mgmt_intf]:
+            if type(ip_network(mgmt_ip, strict=False)) is IPv4Network:
+                mgmt_ipv4 = mgmt_ip.split('/')[0]
+                return intf_mac, mgmt_ipv4
+
+    return intf_mac, mgmt_ipv4
+
+
+@pytest.fixture(scope="module")
+def check_mux_simulator(mux_server_url, ptf_server_intf, tor_mux_intf, ptfadapter, upper_tor_host, lower_tor_host, request, tbinfo):
+    """
+    @summary: Checks if the OVS bridge mux simulator is functioning correctly
+    @return: A dictionary containing the testing result of the PTF interface tested:
+        {
+            'failed': <True/False>,
+            'failed_reason': <reason string>,
+            'intf': '<PTF interface name> mux simulator'
+        }
+    """
+    if request.config.option.skip_sanity or 'dualtor' not in tbinfo['topo']['name']:
+        return {}
+
+    results = {
+                'failed': False,
+                'failed_reason': '',
+                'intf': '{} mux simulator'.format(ptf_server_intf)
+              }
+    logger.info("Checking mux simulator status for PTF interface {}".format(ptf_server_intf))
+    ptf_port_index = int(ptf_server_intf.replace('eth', ''))
+    mux_simulator_index = ptf_port_index + 1
+    recover_all_directions(mux_server_url, ptf_port_index)
+
+    upper_tor_intf_mac, upper_tor_mgmt_ip = get_arp_pkt_info(upper_tor_host, tor_mux_intf)
+    lower_tor_intf_mac, lower_tor_mgmt_ip = get_arp_pkt_info(lower_tor_host, tor_mux_intf)
+
+    upper_tor_ping_tgt_ip = '10.10.10.1'
+    lower_tor_ping_tgt_ip = '10.10.10.2'
```

Comment:
Do these ping target IP addresses need to be configured on PTF interfaces in advance?

Reply:
They don't, sending a `ping` from the ToR results in ARP packets being sent for these IPs which can be verified on the PTF